torch/testing/asserts.py

```diff
@@ -0,0 +1,186 @@
+import functools
+import inspect
+import warnings
+from typing import Callable, Optional, Tuple
+
+import torch
+from torch.testing._internal.common_utils import get_comparison_dtype as _get_comparison_dtype, TestCase as _TestCase
+
+__all__ = ["assert_tensors_equal", "assert_tensors_allclose"]
```

Comment:
assert_array_equal() would analogously be assert_tensor_equal()

https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_array_equal.html#numpy.testing.assert_array_equal

The names are already different because our names have "tensor" in them either way, however, and these names do seem more natural. We might want to add aliases from the NumPy signatures to our test functions in the future, too.

Reply:
> We might want to add aliases from the NumPy signatures to our test functions in the future, too.

So basically `assert_array_allclose` as an alias for `assert_tensors_allclose`?

Reply:
Maybe numpy naming (without the trailing `s`) makes sense, considering that `np.testing.assert_array_equal([0, 0], 0)` is a valid pattern: `assert_array_equal` doesn't necessarily compares 2 arrays.

Reply:
I would prefer we be stricter and require the Python types match, though (with maybe an option to disable that requirement).

Letting `torch.tensor(0) == 0` by default seems likely to be a source of error.